templates/init_runner_script.sh

```diff
@@ -85,6 +85,11 @@ elif [ "${LINUX_OS_VERSION}" = "8" ]
 then
 sudo dnf install -y podman-docker
 DOCKER_SERVICE_START="no"
+elif [ "${LINUX_OS_VERSION}" = "9" ]
```

Comment:
yes. It is meant for runner without the `no-docker` tag. And hence the installation of docker. The script exited because rhel-9 was not supported if you check the code closely.